src/persist-client/src/impl/machine.rs

```diff
@@ -398,14 +412,14 @@ where
         &mut self,
         cmd: &CmdMetrics,
         mut work_fn: WorkFn,
-    ) -> Result<(SeqNo, Result<R, E>), Indeterminate> {
+    ) -> Result<(SeqNo, R, Result<(), Indeterminate>), E> {
```

Comment:
I just kinda rearranged this signature to suit my needs. Feel like I might be missing the principles about which order to nest the `Result`s in?